crates/wasm/wit/component.wit

```diff
@@ -24,6 +28,10 @@ default interface component {
         type-uvec2(list<uvec2>),
         type-uvec3(list<uvec3>),
         type-uvec4(list<uvec4>),
+        type-procedural-mesh-handle(list<procedural-mesh-handle>),
+        type-procedural-texture-handle(list<procedural-texture-handle>),
+        type-procedural-sampler-handle(list<procedural-sampler-handle>),
+        type-procedural-material-handle(list<procedural-material-handle>),
```

Comment:
I don't love having to extend the list of valid value types for every new procedural handle, but I can't think of a better way to do it at present. We'll likely need to revisit our component value model at some point anyway, so this is fine for now, but it is a bit of a shame

Reply:
Agreed. I will nuke these once we have a better model in place.